From: "You-Sheng Yang (vicamo)" <vicamo.yang@canonical.com>
Date: Mon, 19 Sep 2022 19:11:52 +0800
Subject: build: fix kernel feature macro definitions

CONFIG_INTEL_SKL_INT3472 is always enabled as module in Ubuntu kernel >=
5.14, so it's corresponding kernel feature macros are automatically
defined in kernel autoconf.h and should therefore not be redefined
again.

For CONFIG_POWER_CTRL_LOGIC, since it's now obsoleted and disabled by
default, comment-out its feature macro definition as well.

Signed-off-by: You-Sheng Yang (vicamo) <vicamo.yang@canonical.com>
---
 Makefile | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index f4ca9b1..f8bc96b 100644
--- a/Makefile
+++ b/Makefile
@@ -32,11 +32,10 @@ mei_pse-y := drivers/misc/ivsc/mei_pse.o
 obj-m += mei_ace_debug.o
 mei_ace_debug-y := drivers/misc/ivsc/mei_ace_debug.o
 
-export CONFIG_INTEL_VSC = y
+export CONFIG_INTEL_VSC = m
 
 export CONFIG_VIDEO_INTEL_IPU6 = m
 export CONFIG_IPU_ISYS_BRIDGE = y
-export CONFIG_INTEL_SKL_INT3472 = m
 obj-y += drivers/media/pci/intel/
 
 export CONFIG_VIDEO_HM11B1 = m
@@ -58,13 +57,11 @@ ccflags-y += -I$(src)/backport-include/drivers/misc/mei/
 subdir-ccflags-y += -I$(src)/include/
 
 subdir-ccflags-$(CONFIG_INTEL_VSC) += \
-        -DCONFIG_INTEL_VSC
+        -DCONFIG_INTEL_VSC_MODULE=1
 subdir-ccflags-$(CONFIG_IPU_ISYS_BRIDGE) += \
-	-DCONFIG_IPU_ISYS_BRIDGE
-subdir-ccflags-$(CONFIG_INTEL_SKL_INT3472) += \
-	-DCONFIG_INTEL_SKL_INT3472
-subdir-ccflags-$(CONFIG_POWER_CTRL_LOGIC) += \
-	-DCONFIG_POWER_CTRL_LOGIC
+	-DCONFIG_IPU_ISYS_BRIDGE=1
+# subdir-ccflags-$(CONFIG_POWER_CTRL_LOGIC) += \
+# 	-DCONFIG_POWER_CTRL_LOGIC_MODULE=1
 subdir-ccflags-y += $(subdir-ccflags-m)
 
 all:
